boards: add support for makediary m2 devkit

purpl3F0x opened the pull request:
This PR adds support for Makediary NRF52840 M2 devkit

Signed-off-by: Stavros Avramidis <stavros9899@gmail.com>

anangl reviewed (changes requested):
Squash your commits into one, please.

purpl3F0x commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+	pinctrl-names = "default", "sleep";
+};
+
+&spi0 {
+	compatible = "nordic,nrf-spi";
+	status = "okay";
```
The had it that way in their repo,  so i'm not exactly sure about it

thedjnK replied:
This should be spi3

purpl3F0x commented on boards/arm/nrf52840_m2/doc/index.rst:
```diff
@@ -0,0 +1,91 @@
+Overview
+********
+
+`nRF52840 M.2 Developer
+Kit <https://store.makerdiary.com/products/nrf52840-m2-developer-kit>`__
+is a versatile IoT prototyping platform, including the `nRF52840 M.2
+Module <https://store.makerdiary.com/products/nrf52840-m2-module>`__ and
+`M.2 Dock <https://wiki.makerdiary.com/m2-dock>`__. You can use the
```
Not sure if it's that necessary

purpl3F0x committed:
boards: add support for makediary m2 devkit

Signed-off-by: Stavros Avramidis <stavros9899@gmail.com>

carlescufi reviewed (changes requested):
Please add newlines at the end of all files

thedjnK commented on boards/arm/nrf52840_m2/doc/index.rst:
```diff
@@ -0,0 +1,90 @@
+	- Ultra low power 64Mbit QSPI flash memory
+	- U.FL/I-PEX MHF Antenna Connectors
+
+* Integrated DAPLink Debugger
+	- MSC - drag-n-drop programming flash memory
+	- CDC - virtual com port for log, trace and terminal emulation
+	- HID/WEBUSB HID - CMSIS-DAP compliant debug channel
+
+* Advanced Power Profiling
+	- Up to 690 mA current measurement
+	- Voltage tracking
+	- 30000 samples per second
+	- Precision amplifiers with dual gain stage
+	- Cross-platform utility available in Python
```
A module has an embedded daplink and power profiler on it?

purpl3F0x replied:
Yes it does

thedjnK commented on boards/arm/nrf52840_m2/doc/index.rst:
```diff
@@ -0,0 +1,90 @@
+	- Precision amplifiers with dual gain stage
+	- Cross-platform utility available in Python
+
+* Shipped with Python Interpreter firmware
+* 1.3" 240x240 Color IPS TFT LCD Display
+* Li-Po Battery Charger with Power Path Management
+* User programmable LED and Button
+* 2.4 GHz and NFC cabled PCB Antennas
+* microSD Card Slot
+* Arduino Form Factor
+* 4 Grove Connectors
+* Reversible USB-C Connectors
+* 3.3V IO Operating Voltage
```
These are not on the m2 module

purpl3F0x replied:
It's sold as a kit so... I'm just copy pasting their info here

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,316 @@
+	};
+};
+
+
```
Excessive spacing

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2_defconfig:
```diff
@@ -0,0 +1,31 @@
+# Enable hardware stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# Enable RTT
+CONFIG_USE_SEGGER_RTT=y
```
Why or how would someone use segger RTT with a DAPLINK?

purpl3F0x replied:
`pyocd rtt` for example

purpl3F0x commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,316 @@
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <0 0 &gpio0 2 0>, /* A0 */
+		           <1 0 &gpio0 3 0>, /* A1 */
+		           <2 0 &gpio0 28 0>, /* A2 */
+		           <3 0 &gpio0 27 0>, /* A3 */
+		           <4 0 &gpio0 26 0>, /* A4 */
+		           <5 0 &gpio0 4 0>, /* A5 */
+		           <6 0 &gpio0 15 0>, /* D0 */
+		           <7 0 &gpio0 16 0>, /* D1 */
+		           <8 0 &gpio0 19 0>, /* D2 */
+		           <9 0 &gpio0 20 0>, /* D3 */
+		           <10 0 &gpio0 21 0>, /* D4 */
+		           <11 0 &gpio0 22 0>, /* D5 */
+		           <12 0 &gpio0 23 0>, /* D6 */
+		           <13 0 &gpio0 24 0>, /* D7 */
+		           <14 0 &gpio1 0 0>, /* D8 */
+		           <15 0 &gpio1 1 0>, /* D9 */
+		           <16 0 &gpio1 2 0>, /* D10 */
+		           <17 0 &gpio1 3 0>, /* D11 */
+		           <18 0 &gpio1 4 0>, /* D12 */
+		           <19 0 &gpio1 7 0>, /* D13 */
+		           <20 0 &gpio1 5 0>, /* D14 */
+		           <21 0 &gpio1 6 0>;    /* D15 */
+	};
```
I think you're seeing the wrong item
https://wiki.makerdiary.com/nrf52840-m2-devkit/getting-started/

purpl3F0x commented:
Bump to remove stale

thedjnK commented:
> Bump to remove stale

The pending issues haven't been addressed, so it is stale. Are you planning to fix this PR?

purpl3F0x commented:
> scratch_partition

I have addressed most of the issues you mentioned. I know see that I have not replaced the images with jpg and have left a comment on the Kconfig

thedjnK commented:
I can't mark comments as resolved, probably because of some github bug, but you need to fix the issues like the commented code and the format of the files, and do that before anyone else reviews or re-reviews this

purpl3F0x commented:
> I can't mark comments as resolved, probably because of some github bug, but you need to fix the issues like the commented code and the format of the files, and do that before anyone else reviews or re-reviews this

Okay, will do. Also can you check the spi1 pinout configuration ? I think makerdiary had the MISO/MOSI pins wrongly swapped. Not 100% sure tho.

thedjnK commented:
SPI1 is around the wrong way yes, MOSI and MISO are flipped

thedjnK commented:
But also seems unlikely that SPI1 would be used, given that P1.07 is connected to an LED.

purpl3F0x commented:
> But also seems unlikely that SPI1 would be used, given that P1.07 is connected to an LED.

I had not seen that. Seems rather strange cause spi1 also is supposed to be Arduino spi. I guess someone can physically remove the led.

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+		led1 = &led1_red;
+		led2 = &led1_green;
+		led3 = &led1_blue;
+		led0-green = &led0_green;
+		led1-red   = &led1_red;
+		led1-green = &led1_green;
+		led1-blue  = &led1_blue;
```
Not sure these are needed

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+	pinctrl-names = "default", "sleep";
+};
+
+&i2c1 {
+	compatible = "nordic,nrf-twi";
```
Would remove i2c1 since it's on non-standard pins and spi1 can be used instead

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,317 @@
+	cs-gpios = <&arduino_header 16 0>; /* D10 */
+};
+
+&spi2 {
+	compatible = "nordic,nrf-spi";
+	/* Cannot be used together with i2c1. */
+	// status = "okay";
```
Unresolved

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+	pinctrl-0 = <&spi0_default>;
+	pinctrl-1 = <&spi0_sleep>;
+	pinctrl-names = "default", "sleep";
+	cs-gpios = <&gpio0 6 0>,<&gpio0 7 0>;
```
Separate with space after comma, plus these are missing GPIO flags

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+		compatible = "zephyr,mmc-spi-slot";
+		reg = <1>;
+		status = "okay";
+		label = "SDHC0";
```
Labels are no longer used

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+	};
+
+	sdhc0: sdhc@1 {
+		compatible = "zephyr,mmc-spi-slot";
```
This driver was removed ages ago, should be using `zephyr,sdhc-spi-slot` and needs to follow the correct dts layout for it

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+		backlight-pwm-led = &backlight_pwm_led;
+		watchdog0 = &wdt0;
+		spi-flash0 = &mx25r64;
+	};
```
Missing mcuboot/bootloader aliases for led and button

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+};
+
+
+&qspi {
```
Seems a bit odd to have an 8MB QSPI flash but not bother using it for any flash partitions?

purpl3F0x replied:
hmmm, I would guess it's primary usage will be for data logging/ file storage. So maybe put the settings partition in there ?

thedjnK commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+#include "nrf52840_m2-pinctrl.dtsi"
+
+/ {
+	model = "Makerdiary nRF52840-M2 Module";
```
This is not correct, also the name of the board should be changed, nrf52840-m2 is the name of the module, the name of the board/devkit is nrf52840 m2 developer kit

thedjnK commented on boards/arm/nrf52840_m2/doc/index.rst:
```diff
@@ -0,0 +1,90 @@
+.. _nrf52840_m2:
```
Outstanding issue, images need to be .png or .jpg and no more than 40KB per image

purpl3F0x commented on boards/arm/nrf52840_m2/nrf52840_m2.dts:
```diff
@@ -0,0 +1,309 @@
+		};
+		led1_green: led_2 {
+			gpios = <&gpio0 29 GPIO_ACTIVE_LOW>;
+			label = "Green LED 1";
+		};
+		led1_blue: led_3 {
+			gpios = <&gpio0 31 GPIO_ACTIVE_LOW>;
+			label = "Blue LED 1";
```
This is correct it's an RGB led, at least there is a logic behind all being led 1
